project2/screens/search.js

```diff
@@ -1,131 +1,72 @@
-
-
-const Row = props => (
-    <TouchableOpacity onPress={() => props.selected(props.id)}>
-        <View style={styles.row}>
-            <Text style={styles.listText}>{props.title}</Text>
-        </View>
-    </TouchableOpacity>
-)
-const renderItem = selected => obj => <Row selected={selected}{...obj.item} />
-
-export default class search extends React.Component {
-    static navigationOptions = {
-        title: "Search",
-        headerTitleStyle: { color: "#AAAAAF", }
-    };
-
-    state = {
-        searchString: "",
-        movies: "",
-        totalResults: 0,
-        totalPages: 0,
-        nextPage: 0,
-    };
-
-    getMovies = async (searchString) => {
-        const results = await searchMovies(`${searchString}`)
-        console.log(results)
-
-        if (results !== undefined) {
-            this.setState({ movies: results["result"] })
-            this.setState({ totalResults: results["totalResults"] })
-            this.setState({ totalPages: results["totalPages"] })
-
-            if (results["totalPages"] > 1) {
-                this.setState({ nextPage: 2 })
-            }
-        }
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.searchString !== prevState.searchString) {
-            this.getMovies(this.state.searchString)
+import SearchListItem from "./Search/ListItem";
+import s from "../Styles";
+
+const Search = () => {
+  const [searchString, setSearchString] = React.useState("");
+  const [movies, setMovies] = React.useState([]);
+  const [totalResults, setTotalResults] = React.useState(0);
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState(0);
+
+  React.useEffect(() => {
+    const getMovies = async () => {
+      const { result, totalResults, totalPages } = await searchMovies(
+        searchString
+      );
+
+      if (result) {
+        setMovies(result);
+        setTotalResults(totalResults);
+        setTotalPages(totalPages);
+
+        if (totalPages > 1) {
+          setNextPage(2);
         }
-    }
-
-    updateSearch = searchString => {
-        this.setState({
-            searchString: searchString,
-            movies: "",
-            totalResults: 0,
-            totalPages: 0,
-            nextPage: 0
-        })
-    }
-
-    movieSelected = (id) => {
-        this.props.navigation.navigate("Movie", { "MovieID": id })
-    }
-
-    displayMore = async () => {
-        if (this.state.totalPages === 1) return
-        if (this.state.nextPage === this.state.totalPages) return
-
-        const results = await getMoviePage(this.state.searchString, this.state.nextPage)
-
-        if (results !== undefined) {
-            this.setState(prevState => ({ movies: [...prevState.movies, ...results] }));
+      }
+    };
 
-            if (this.state.nextPage < this.state.totalPages) {
-                this.setState({ nextPage: this.state.nextPage++ })
-            }
-        }
+    if (searchString !== "") {
```

Comment:
We only want to actually run a search if the search string isn't empty. This doesn't actually matter here as there isn't a search until we hit three characters, but anyway